src/controllers/admin/DebugController.php

```diff
@@ -40,8 +44,20 @@ public function displayAjaxSetStatus()
             PacklinkPrestaShopUtility::die400();
         }
 
-        $this->getConfigService()->setDebugModeEnabled($data['status']);
+        $this->baseController->setStatus($data['status']);
 
         PacklinkPrestaShopUtility::dieJson(array('status' => $data['status']));
     }
+
+    /**
+     * Downloads system info zip file.
+     *
+     * @throws \PrestaShopException
+     */
+    public function displayAjaxGetSystemInfo()
+    {
+        $file = SystemInfoUtility::getSystemInfo();
```

Comment:
The reason why the BaseDebugController implementation is not used is that the controller in the PS integration cannot extend the Core Debug controller and just override the getIntegrationInfo method implementation because it already extends the PS admin controller. Therefore, it cannot extend the base method but needs to have its own implementation of that method. The same has been done in the WooCommerce integration.
I propose that we have an interface in the Core which should be extended by a service in each integration that will return system-specific information and then that information can be combined along with the general data that is the same for all integrations into one system info archive.

Reply:
You could just have another controller (or that system info utility) extend the core controller and use it here. That way, the code would not be duplicated.